src/script/lua_api/l_object.cpp

```diff
@@ -235,8 +235,14 @@ int ObjectRef::l_set_hp(lua_State *L)
 		return 0;
 
 	// Get HP
+	int old_hp = co->getHP();
 	int hp = lua_tonumber(L, 2);
 
+	// Protect from damage if immortal
+	if ((itemgroup_get(co->getArmorGroups(), "immortal") == 1) && (hp < old_hp)) {
```

Comment:
can't you use co->isImmortal() here?

Reply:
Yes, that's what I intend to do. This PR _should_ be ready for review by the evening.

Reply:
Done